client/txn.go

```diff
@@ -445,5 +461,26 @@ func (txn *Txn) send(reqs ...roachpb.Request) (*roachpb.BatchResponse, *roachpb.
 			txn.Proto.Status = roachpb.ABORTED
 		}
 	}
+
+	// If we inserted a begin transaction request, remove it here.
+	if needBeginTxn {
+		if br != nil && br.Responses != nil {
+			br.Responses = append(br.Responses[:firstWriteIndex], br.Responses[firstWriteIndex+1:]...)
+		}
+		// Handle case where inserted begin txn confused an indexed error.
+		if pErr != nil && pErr.Detail != nil {
+			if iErr, ok := pErr.Detail.GetValue().(roachpb.IndexedError); ok {
+				if idx, ok := iErr.ErrorIndex(); ok {
+					if idx == int32(firstWriteIndex) {
+						// An error was encountered on begin txn; disallow the indexing.
+						pErr = roachpb.NewError(util.Errorf("error on begin transaction: %s", pErr.Detail.GetValue().(error)))
```

Comment:
I'm just formatting with pErr. Didn't want to use `pErr.GoError()` because that would give an index, which would confuse callers, as BeginTransaction is added to the batch and then removed.